api/envoy/config/core/v3/protocol.proto

```diff
@@ -280,8 +280,24 @@ message Http2ProtocolOptions {
   // the whole HTTP/2 connection is terminated upon receiving invalid HEADERS frame. However,
   // when this option is enabled, only the offending stream is terminated.
   //
+  // This is overridden by HCM :ref:`stream_error_on_invalid_http_messaging
+  // <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.upgrade_configs>`
+  // iff present.
+  //
+  // This is deprecated in favor of stream_error_on_invalid_http_message.
+  //
+  // See `RFC7540, sec. 8.1 <https://tools.ietf.org/html/rfc7540#section-8.1>`_ for details.
+  bool stream_error_on_invalid_http_messaging = 12 [deprecated = true];
+
+  // Allows invalid HTTP messaging and headers. When this option is disabled (default), then
+  // the whole HTTP/2 connection is terminated upon receiving invalid HEADERS frame. However,
+  // when this option is enabled, only the offending stream is terminated.
+  //
+  // This overrides any HCM :ref:`stream_error_on_invalid_http_messaging
+  // <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.upgrade_configs>`
+  //
   // See `RFC7540, sec. 8.1 <https://tools.ietf.org/html/rfc7540#section-8.1>`_ for details.
-  bool stream_error_on_invalid_http_messaging = 12;
+  google.protobuf.BoolValue stream_error_on_invalid_http_message = 14;
```

Comment:
How about `stream_error_on_invalid_http_messaging_hcm_override`? That's very very long, but I worry about the similarity in names between new and deprecated.

Reply:
Yeah, I agree on name similarity but I'd mildly prefer a clean name in the long run.  @mattklein123 WDYT?

Reply:
Meet half way on `stream_error_on_invalid_http_messaging_override`? :)

Reply:
Haha, deal, modulo messaging -> message for consistency with the long term HCM value.

Alternately would you also be OK with override_stream_error_on_invalid_http_message?  I think that reads a bit better but it's arguably easier to overlook the difference so I'm fine if you prefer stream_error_on_invalid_http_message_override

Reply:
Yep, either is fine.